### PR TITLE
refactor (usb): Restructure code organization

### PR DESCRIPTION
**Pull request by ujjwal-cyph**

### CODE REVIEW

File: common/interfaces/desktop_app_interface/usb_api_priv.h [hunk 1]
```diff
@@ -16,22 +16,189 @@
...
+} comm_status_t;
+
 /*****************************************************************************
  * EXPORTED VARIABLES
  *****************************************************************************/
 
+// received/ready-to-send cmd can be decoded/encoded into this buffer
+extern uint8_t comm_io_buffer[COMM_BUFFER_SIZE];
+extern comm_payload_t comm_payload;
+extern comm_status_t comm_status;
```
Comment (amanCypherock) [inline common/interfaces/desktop_app_interface/usb_api_priv.h:159]: Preferred to expose these variables through APIs in this file. This is according to the defined coding guidelines.

File: common/interfaces/desktop_app_interface/usb_api_priv.h [hunk 2]
```diff
@@ -16,22 +16,189 @@
...
+
 /*****************************************************************************
  * GLOBAL FUNCTION PROTOTYPES
  *****************************************************************************/
+
+static inline void comm_set_payload_struct(uint16_t proto_len,
```
Comment (amanCypherock) [inline common/interfaces/desktop_app_interface/usb_api_priv.h:165]: ATCG: Inline functions are not preferred as they are harder to mock for unit tests. Suggest moving the function definition to source file.

File: common/interfaces/desktop_app_interface/usb_rx_parser.c [hunk 3]
```diff
@@ -0,0 +1,296 @@
+/**
+ * @file    usb_rx_parser.c
```
Comment (kartikCypherock) [inline common/interfaces/desktop_app_interface/usb_rx_parser.c:2]: consider renaming to `comm_rx_parser.c` as it is independent of physical layer

File: common/interfaces/desktop_app_interface/usb_internals.c [hunk 4]
```diff
@@ -0,0 +1,479 @@
+/**
+ * @file    usb_internal.c
```
Comment (kartikCypherock) [inline common/interfaces/desktop_app_interface/usb_internals.c:2]: consider renaming to `comm_internal.c` as it is independent of physical layer

File: common/interfaces/desktop_app_interface/usb_rx_parser.c (hunk 3 above)
Comment (kartikCypherock) [reply common/interfaces/desktop_app_interface/usb_rx_parser.c:2]: not valid because we are calling lusb functions, therefore this comment is redundant

**Review by kartikCypherock — APPROVED** [review]

File: common/interfaces/desktop_app_interface/usb_api_priv.h (hunk 1 above)
Comment (ujjwal-cyph) [reply common/interfaces/desktop_app_interface/usb_api_priv.h:159]: will take this in next change request

Comment (amanCypherock) [reply common/interfaces/desktop_app_interface/usb_api_priv.h:159]: Resolving as extern declared only in _priv header. But for future changes, we should not allow such usage, deviating from coding guidelines.

File: common/interfaces/desktop_app_interface/usb_api_priv.h (hunk 2 above)
Comment (amanCypherock) [reply common/interfaces/desktop_app_interface/usb_api_priv.h:165]: Resolving as not introduced in this pr and defined in _priv file

**Review by amanCypherock — APPROVED** [review]

File: common/interfaces/desktop_app_interface/usb_api_priv.h (hunk 1 above)
Comment (amanCypherock) [reply common/interfaces/desktop_app_interface/usb_api_priv.h:159]: Preferred to change this PR as introduced here only

Comment (ujjwal-cyph) [reply common/interfaces/desktop_app_interface/usb_api_priv.h]: chnaged

**Pull request merged**